test/rules/no-use-before-declare/test.ts.lint

```diff
@@ -68,7 +74,8 @@ export {
 };
 
 var undeclaredA = 42;
-let { undeclaredB } = { undeclaredB: 43 };
+var { undeclaredB } = { undeclaredB: 43 };
+var { undeclaredB: undeclaredA } = { undeclaredB: 43 };
 const [ undeclaredC, [undeclaredD] ] = [ 1, [2] ];
 
```

Comment:
Can we add test cases for multiple reassignments and reassignment and non-reassignment in the same destructuring?